src/App.jsx

```diff
@@ -67,10 +76,54 @@ class App extends Component {
     });
   }
 
+  GPSTimerOut = () => {
+    clearTimeout(Timer.gameTimer);
+    this.setState({ locationOk: LocationStatus.locationTimerOut });
+    this.restartQuestions();
+  }
+
+  continiusGPSCheck = () => {
+    const {
+      continusGPSChckerTime,
+    } = this.state;
+    continusGPSChckerTimer = setTimeout(
+      this.continiusGPSCheck,
+      continusGPSChckerTime,
+    );
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
```

Comment:
**Suggestion:** Is there a reason `App` handles geolocation? It seems more suiting if the `GPSCheck` component takes care of this. Look into `this.props.children` and [composition](https://reactjs.org/docs/composition-vs-inheritance.html).

We could then write the structure of our app in this component, but be sure it only renders when the user is in an accepted location as follows:

```html
<GPSCheck locationOk={locationOk}>
  ... the rest of the app ...
</GPSCheck>
```